v1/ao/internal/reporter/oboe.go

```diff
@@ -163,9 +163,7 @@ func sendInitMessage() {
 func (b *tokenBucket) count(sampled, hasMetadata, rateLimit bool) bool {
 	b.RequestedInc()
 
-	if hasMetadata {
-		b.ThroughInc()
-	} else {
+	if !hasMetadata {
```

Comment:
So I assume you are fixing the logic, so it would NOT increment ThroughTraceCount if the incoming metadata has sampled bit set as `0`? :)

Reply:
it's correct.